solr/core/src/java/org/apache/solr/search/grouping/distributed/responseprocessor/SearchGroupShardResponseProcessor.java

```diff
@@ -135,7 +135,7 @@ public void process(ResponseBuilder rb, ShardRequest shardRequest) {
           shards.add(srsp.getShard());
         }
       }
-      hitCountDuringFirstPhase += (Integer) srsp.getSolrResponse().getResponse().get("totalHitCount");
+      hitCountDuringFirstPhase += ((Number) srsp.getSolrResponse().getResponse().get("totalHitCount")).intValue();
```

Comment:
Will this need a null check as well?

Reply:
Actually. The old code would have NPEd if this were null. Since `hitCountDuringFirstPhase` is an `int` and it would have unboxed here.

So ignore my comment!